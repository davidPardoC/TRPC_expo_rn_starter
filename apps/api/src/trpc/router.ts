import { UserSchema } from "common/schemas/user.schema";
import { publicResource, router } from ".";
import { USerServices } from "../services/user.service";

const userServices = new USerServices();

export const appRouter = router({
  getUsers: publicResource.query(async ()=>{
   return userServices.getUsers()
  }),

  createUser: publicResource.input(UserSchema).mutation(async (opts) => {
    const { input } = opts;
    const user = await userServices.createUser(input);
    return user
  }),
});

export type AppRouter = typeof appRouter;
