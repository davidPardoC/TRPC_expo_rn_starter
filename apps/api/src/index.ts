import express from "express";
import * as trpcExpress from '@trpc/server/adapters/express';
import { appRouter } from "./trpc/router";

const app = express();
const PORT = process.env.PORT || 5000;

const createContext = ({req, res}: trpcExpress.CreateExpressContextOptions) => ({})

app.use("/status", (_req, res) => {
  res.json({ status: "ok" });
});

app.use("/trpc", trpcExpress.createExpressMiddleware({router: appRouter, createContext }))

app.listen(PORT, () => {
  console.log(`âœ… Server started un http://localhost:${PORT}`);
});
