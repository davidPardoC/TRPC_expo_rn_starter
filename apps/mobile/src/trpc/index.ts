import { createTRPCProxyClient, httpBatchLink } from "@trpc/client";
import {AppRouter} from '../../../api/src/trpc/router'

export const trpc = createTRPCProxyClient<AppRouter>({
    links: [httpBatchLink({url:'http://10.0.2.2:5000/trpc'})]
})
