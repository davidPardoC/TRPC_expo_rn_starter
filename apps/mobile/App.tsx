import { useEffect, useState } from "react";
import { Button, FlatList,  SafeAreaView,  StatusBar,  StyleSheet, Text } from "react-native";
import { trpc } from "./src/trpc";

export default function App() {
  const [users, setUsers] = useState<{ email: string; password: string }[]>([]);
  
  const getUsers = async () => {
    const data = await trpc.getUsers.query();
    setUsers(data);
  };

  useEffect(() => {
    getUsers();
  }, []);

  const createUser = async () => {
    await trpc.createUser.mutate({
      email: "some@email.com",
      password: "some-password",
    });
    getUsers();
  };

  return (
    <SafeAreaView style={styles.container}>
      <Text>TRPC REACT NATIVE EXPO STARTER!</Text>
      <StatusBar  />
      <Button title="Create User" onPress={createUser} />
      <FlatList
        data={users}
        renderItem={({ item: { email } }) => <Text style={styles.listItem}>{email}</Text>}
        style={styles.listContainer}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: 'center',
    justifyContent: 'center',
  },
  listItem : {
    padding: 10
  },
  listContainer:{
    padding: 10
  }
});
