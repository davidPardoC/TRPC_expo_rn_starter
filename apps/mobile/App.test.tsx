import { render, screen, fireEvent, act } from "@testing-library/react-native";
import App from "./App";
import { trpc } from "./src/trpc";

jest.mock("./src/trpc", () => ({
  trpc: { getUsers: { query: jest.fn() }, createUser: { mutate: jest.fn() } },
}));

describe("<App/>", () => {
  it("App snapshot", async () => {
    const component = render(<App />);

    await act(async () => {
      expect(component).toMatchSnapshot();
    });
  });

  it("Should call trpc createUser", async () => {
    const spy = jest
      .spyOn(trpc.createUser, "mutate")
      .mockResolvedValueOnce({
        email: "test-email",
        password: "test-password",
      });
    const { getByRole } = render(<App />);

    await act(async () => {
      const btn = getByRole("button");
      fireEvent.press(btn);
    });

    expect(spy).toHaveBeenCalledWith({
      email: "some@email.com",
      password: "some-password",
    });
  });
});
