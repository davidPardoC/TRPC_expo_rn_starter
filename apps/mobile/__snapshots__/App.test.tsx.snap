// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> App snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    TRPC REACT NATIVE EXPO STARTER!
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Create User
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "padding": 10,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
